src/components/Calculator.js

```diff
@@ -1,41 +1,81 @@
 import Number from "./Number";
 import Operation from "./Operation";
 import Screen from "./Screen";
+import { useState } from 'react';
 
 const Calculator = () => {
+
   /** TODO: Here is where you are going to keep track of calculator state */
+  const [count, setCount] = useState(0);
+  const [displayCount, setDisplayCount] = useState(0);
+  var operator;
 
   /** TODO: what happens when I click a number? */
-  const handleNumberClick = () => {};
+  const handleNumberClick = (value) => {
+    if (operator === "+") {
+      setCount(parseInt(count) + value);
+      setDisplayCount(value);
+    } else if (operator === "-") {
+      setCount(parseInt(count) - value);
+      setDisplayCount(value);
+    } else if (operator === "x") {
+      setCount(parseInt(count) * value);
+      setDisplayCount(value);
+    } else if (operator === "/") {
+      setCount(parseInt(count) / value);
+      setDisplayCount(value);
+    } else {
+      if (parseInt(count) === 0) {
+        setCount(value);
+        setDisplayCount(value);
+      } else {
+        setCount(parseInt('' + count + value));
+        setDisplayCount(parseInt('' + count + value));
+      }
+    }
+  };
 
   /** TODO: what happens when I click an operation? */
-  const handleOperationClick = () => {};
+  const handleOperationClick = (value) => {
+    if (value === "clear") {
+      setCount(0);
+      setDisplayCount(0);
+    } else if (value === "=") {
+      setDisplayCount(count);
+    } else {
+      operator = value;
+    }
+  };
 
   return (
-    <div>
-      <Screen value="123" />
-      <div style={{ display: "flex" }}>
-        <div>
+    <div style={{ border: "5px solid #02040F", borderRadius: "10px", padding: "0.7rem", backgroundColor: "#f2f2f2",}}>
+      <Screen value={displayCount} />
+      <div style={{ display: "flex", gap: "0.6rem" }}>
+        <div style={{ flexBasis: "33%", flexGrow: "1" }}>
           <Number value={0} onClick={handleNumberClick} />
```

Comment:
`[0,1,2,3,4].map(n => <Number key={n} value={n} onClick={handleNumberClick} />`

Reply:
Hi Nico, I cleaned this up with .map() like you suggested, thank you!

When I run the app locally, I'm getting the console Warning in the attached screenshot. But I do not see any lists in my app where I don't have the key prop.

And when I view the app on the published Github page, the console does not show the warning, even though my local is completely synced up. Is that expected?

![Screen Shot 2022-10-12 at 11 36 22 PM](https://user-images.githubusercontent.com/54754859/195493962-4fe80463-d06b-4b97-82b4-174a5e0b362d.png)

Lastly, when I click any number repeatedly, it messes up on the 17th click. Im reading online that integers can only be accurate up to a certain number of digits. Could that be a factor?

Thank you for your help!

Reply:
The problem is your `makeKeys` function. In the signature, you destruct the array, which then forces you to iterate over 5 items even if not always 5 are provided (for example when you render the `'clear'` key. A simpler way to write that function would be:

```
const makeKeys = (type, keysArr) => (
    <div style={{ flexBasis: "33%", flexGrow: "1" }}>
      {keysArr.map((n) => {
        if (type === "number") {
          return <Number key={n} value={n} onClick={handleNumberClick} />;
        } else {
          return <Operation key={n} value={n} onClick={handleOperationClick} />;
        }
      })}
    </div>
  );
```